### ISSUE TITLE
Exception when uploading event: Supplied value is not a correct Date

### ISSUE THREAD

**Issue by mhoff** [sabre-io/dav#1016]
Hello, I ran into a problem syncing calendars in nextcloud with evolution as CalDAV client. To be precise, an UnsupportedMediaType-Exception is being thrown [here](https://github.com/sabre-io/dav/blob/master/lib/CalDAV/Plugin.php#L846) when trying to upload the following calendar:

```
BEGIN:VCALENDAR
CALSCALE:GREGORIAN
PRODID:-//Ximian//NONSGML Evolution Calendar//EN
VERSION:2.0
BEGIN:VTIMEZONE
TZID:/freeassociation.sourceforge.net/Europe/Berlin
X-LIC-LOCATION:Europe/Berlin
BEGIN:STANDARD
TZNAME:CET
DTSTART:19701029T030000
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=10
TZOFFSETFROM:+0200
TZOFFSETTO:+0100
END:STANDARD
BEGIN:DAYLIGHT
TZNAME:CEST
DTSTART:19700326T020000
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=3
TZOFFSETFROM:+0100
TZOFFSETTO:+0200
END:DAYLIGHT
END:VTIMEZONE
BEGIN:VEVENT
UID:6ec2c1780c50d18a344615aae6bb61843a14ea0d
DTSTAMP:20171118T121717Z
DTSTART;VALUE=DATE;TZID=/freeassociation.sourceforge.net/Europe/Berlin: 20171117T000000
DTEND;VALUE=DATE;TZID=/freeassociation.sourceforge.net/Europe/Berlin: 20171117T204800
SEQUENCE:4
SUMMARY:Some title
TRANSP:OPAQUE
CLASS:PUBLIC
CREATED:20171119T152718Z
LAST-MODIFIED:20171119T195928Z
LOCATION:Some location
END:VEVENT
END:VCALENDAR  
```

The reported error is:
`Sabre\DAV\Exception\UnsupportedMediaType: Validation error in iCalendar: The supplied value (20171117T000000) is not a correct DATE`

Hence, this entry of the above calendar appears to be broken:
`DTSTART;VALUE=DATE;TZID=/freeassociation.sourceforge.net/Europe/Berlin: 20171117T000000`

Is it?

**Comment by brunt82** [comment]
Have a look at: http://sabre.io/vobject/usage_2/

In "Date and time handling" it is described, that TZID needs a VTIMEZONE object:

> When a TZID is specified, there should also be a matching VTIMEZONE object with all the timezone information. VObject cannot currently automatically embed this. However, in reality other clients seem to do fine without this information. Yet, for completeness, this will be added in the future.

I suppose this is your problem. Maybe another client may work. Have you already tried Thunderbird?

**Comment by evert** [comment]
In this case the error is actually the date format. ` 20171117T000000` is definitely wrong for VALUE=DATE. It contains an extra space, and it's not supported to include time. This would be the correct format: `20171117`. Which evolution version is this?

**Comment by mhoff** [comment]
@brunt82 
> Have you already tried Thunderbird?

Yes, but I had some problems with Thunderbird (+ Lightning) in the past and this is why I gave evolution a try.

@evert 
> Which evolution version is this?

3.26.1-1 -- as far as I can see this is the most recent version for Ubuntu 17.10.

**Comment by evert** [comment]
I would consider submitting a ticket to the evolution project then @mhoff . It's definitely a bug on their end.
